Compile ASPECT main branch with deal.ii 9.6

When I compiled ASPECT main branch with deal.ii 9.6, I have this error:

    In file included from /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/base/array_view.h:20,
                     from /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/base/mpi.h:20,
                     from /quobyte/billengrp/lochy/Software/aspect/include/aspect/global.h:28,
                     from /quobyte/billengrp/lochy/Software/aspect/build/CMakeFiles/aspect.exe.release.dir/cmake_pch.hxx:5,
                     from <command-line>:
    /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/fe/component_mask.h:236:3: error: expected class-name before ‘{’ token
      236 |   DeclExceptionMsg(ExcNoComponentSelected,
          |   ^~~~~~~~~~~~~~~~
    /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/fe/component_mask.h:236:3: error: ‘virtual void aspect::dealii::ComponentMask::ExcNoComponentSelected::print_info(std::ostream&) const’ marked ‘override’, but does not override
      236 |   DeclExceptionMsg(ExcNoComponentSelected,
          |   ^~~~~~~~~~~~~~~~
    In file included from /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/base/patterns.h:27,
                     from /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/base/parameter_handler.h:22,
                     from /quobyte/billengrp/lochy/Software/aspect/include/aspect/plugins.h:28,
                     from /quobyte/billengrp/lochy/Software/aspect/build/CMakeFiles/aspect.exe.release.dir/cmake_pch.hxx:6:
    /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/fe/component_mask.h:180:27: error: ‘numbers’ has not been declared
      180 |                           numbers::invalid_unsigned_int) const;
          |                           ^~~~~~~
    /quobyte/billengrp/Software/deal.ii/deal.ii-9.6.1-toolchain-gcc-13.2.0-openmpi5.0.5/deal.II-v9.6.1/include/deal.II/fe/component_mask.h:190:30: error: ‘numbers’ has not been declared
      190 |                              numbers::invalid_unsigned_int) const;
          |                              ^~~~~~~
    
    
My collaborator has compiled this a few days earlier, so I just roll back my ASPECT to a previous commit 2 days ago 52d7ada1bf08ea672cd38871b91534cf897ff458. 
And it succeeds. Each time I compile, I would delete the old build directory and restart, so the build should be clean.

@anne-glerum has reported the same issue  on Elements. 

The first error message is
```
/scratch/usr/bbpanneg/software/dealii/RL9_candi_9.6.0/deal.II-v9.6.0/include/deal.II/fe/component_mask.h:236:3: error: expected class-name before ‘{’ token
  236 |   DeclExceptionMsg(ExcNoComponentSelected,
      |   ^~~~~~~~~~~~~~~~
/scratch/usr/bbpanneg/software/dealii/RL9_candi_9.6.0/deal.II-v9.6.0/include/deal.II/fe/component_mask.h:236:3: error: ‘virtual void aspect::dealii::ComponentMask::ExcNoComponentSelected::print_info(std::ostream&) const’ marked ‘override’, but does not override
  236 |   DeclExceptionMsg(ExcNoComponentSelected,
      |   ^~~~~~~~~~~~~~~~

```
Note that it talks about `aspect::dealii::ComponentMask`. That's definitely the wrong class name. And it reports on it in a deal.II file. Other error messages talk about `aspect::std::unordered_set`. I have the suspicion that at some place or other, we forget to close `namespace aspect { ...` with your configuration, and that that gets us in trouble.

Two suggestions:

* Can you narrow down further which patch broke things? I'm not sure about your commit numbers above since the second one you mention is from before the other one, if I see this right. Either way, there are only a few commits inbetween.
* What happens if you disable both UNITY builds and preprocessed headers? What is the first error message in that case?

--- a/include/aspect/compat.h
+++ b/include/aspect/compat.h
@@ -206,7 +206,6 @@ namespace aspect
        */
       const PolarManifold<spacedim> polar_manifold;
   };
-}
 
 #else
 
@@ -226,8 +225,6 @@ namespace aspect
 // that we would like to use also for earlier versions
 #if !DEAL_II_VERSION_GTE(9,7,0)
 
-namespace aspect
-{
   using namespace dealii;
 
   /**

